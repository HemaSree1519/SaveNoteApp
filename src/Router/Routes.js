import React from "react";
import {Route, Switch} from "react-router-dom";
import Home from "../Containers/Home/index";
import Login from "../Containers/Login/index";
import SignUp from "../Containers/SignUp/index";
import NotesBoard from "../Containers/NotesBoard/index"
import NotFound from "../Containers/PageNotFound/index";
import AppliedRoute from "./AppliedRoute";

export default ({childProps}) =>
    <Switch>
        <Route path="/" exact component={Home}/>
        <Route path="/notes" exact component={NotesBoard}/>
        <AppliedRoute path="/signup" exact component={SignUp} props={childProps}/>
        <AppliedRoute path="/login" exact component={Login} props={childProps}/>
        <Route component={NotFound}/>
    </Switch>;