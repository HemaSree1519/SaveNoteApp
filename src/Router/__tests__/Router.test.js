import React from 'react';
import { shallow } from 'enzyme';
import Router from '../Routes';
describe('Router', () => {
    it('should render correctly', () => {
        const component = shallow(<Router/>);
        expect(component).toMatchSnapshot();
    });
});