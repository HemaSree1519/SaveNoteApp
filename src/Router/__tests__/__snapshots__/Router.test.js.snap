exports[`Router should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Route
          component={[Function]}
          exact={true}
          path="/" />,
        <Route
          component={[Function]}
          exact={true}
          path="/notes" />,
        <Unknown
          component={[Function]}
          exact={true}
          path="/signup"
          props={undefined} />,
        <Unknown
          component={[Function]}
          exact={true}
          path="/login"
          props={undefined} />,
        <Route
          component={[Function]} />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/notes",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/signup",
          "props": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/login",
          "props": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/" />,
          <Route
            component={[Function]}
            exact={true}
            path="/notes" />,
          <Unknown
            component={[Function]}
            exact={true}
            path="/signup"
            props={undefined} />,
          <Unknown
            component={[Function]}
            exact={true}
            path="/login"
            props={undefined} />,
          <Route
            component={[Function]} />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/notes",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/signup",
            "props": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/login",
            "props": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
